File: sem 3/DS/sorting/selection.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<int> nums = {5, 3, 6, 7, 2, 9, 10, 14, 4};
    for (int i = 0; i<nums.size(); i++) {
        int shortest = i;
        for (int j = i+1; j < nums.size(); j++ ) {
            if (nums[j] < nums[shortest]) {
                shortest = j;
            }
        }
        swap(nums[shortest], nums[i]);
    }

    for (int i=0; i<nums.size(); i++) {
        cout << nums[i] << " ";
    }
    cout << endl; 
    return 0;
}

File: sem 3/DS/sorting/selection.py
nums = [5, 3, 6, 7, 2, 9, 10, 14, 4]

for i in range(len(nums)):
    shortest = i
    for j in range(i+1, len(nums)):
        if nums[j] < nums[shortest]:
            shortest = j
    nums[i], nums[shortest] = nums[shortest], nums[i]

print(nums)